src/components/UserDetailSections/PatronBlock/PatronBlock.js

```diff
@@ -98,33 +71,35 @@ class PatronBlock extends React.Component {
     };
   }
 
-  componentDidMount() {
+  toggleSectionOnBlocksLoaded() {
     const {
-      mutator: {
-        manualPatronBlocks,
-      },
-      user,
+      arePatronBlocksLoaded,
+      patronBlocks,
       onToggle,
-      expanded,
       accordionId,
-      automatedPatronBlocks,
+      expanded,
     } = this.props;
-    const query = `userId==${user.id}`;
-
-    manualPatronBlocks.reset();
-    manualPatronBlocks.GET({ params: { query } })
-      .then(records => {
-        const blocks = records.filter(p => moment(moment(p.expirationDate).endOf('day')).isSameOrAfter(moment().endOf('day')));
-        if ((blocks.length > 0 && !expanded) || (!blocks.length && expanded)) {
-          onToggle({ id: accordionId });
-        }
-      });
+    const { isToggledOnBlocksLoaded } = this.state;
+
+    if (!isToggledOnBlocksLoaded && arePatronBlocksLoaded) {
+      if (!expanded && patronBlocks.length) {
```

Comment:
If we should not toggle patron blocks if user already clicked "**Expand all**" button we have to pass additional props here which says that user clicked "**Expand all**" button and we have to prevent toggling in this place.